core/templates/dev/head/domain/topic/TopicRightsObjectFactory.ts

```diff
@@ -80,15 +80,11 @@ export class TopicRightsObjectFactory {
   // This function takes a JSON object which represents a backend
   // topic python dict.
   createFromBackendDict(
-      topicRightsBackendObject: {
-        published: boolean;
-        canPublishTopicBool: boolean;
-        canEditTopicBool: boolean;
-      }) {
+      topicRightsBackendObject: any) {
```

Comment:
This concerns me. Why does typing raise errors?

Reply:
This is because if I type it then `can_publish_topic` would raise a lint error, so I changed it to camelCase. But this also needs to changed everywhere it is being referenced and most importantly in all of the backend since it is passed as a dict with these keys. So, this comes under 'complex' criteria and I decided to type it as `any`

I think we should file an issue to overcome this. Note that this might be tricky because I think pylint has a rule against naming in camelCase and favors underscore_case.

What do you think?

Reply:
Yes, I understand the complexity. One thing I'd say is -- for anything like this, please file an issue proactively, don't wait for a reviewer to point it out. Thanks!

Reply:
Okay I will keep that in mind in the future.
Raised #7176,
Thanks!